### PR TITLE
[MDS-6150]

### PR DESCRIPTION
**Pull request by taraepp**
## Objective 
- be able to submit a primary document for a major mine application.
- and also be able to go back and add a supporting document if changes are requested
- TS the MMA page component
- brought the form up to this component so all sub-components will have nice consistent access to form data and submit can be triggered by putting `htmlType='submit'` on the button
- there must be a primary document in order to submit, but there does not need to be a *freshly uploaded* primary document anymore.
- combined the "Steps" in with the component itself
- got rid of a couple of warnings spamming my console, except for the MajorMineApplicationReviewSubmit messages, because there's a separate ticket to fix a bug on that component
- the test is a "new" test but it's no less useless than it was.
  - might implement Mat's new test thing tomorrow to make it useful.

[MDS-6150](https://bcmines.atlassian.net/browse/MDS-6150)

_Why are you making this change? Provide a short explanation and/or screenshots_

### CODE REVIEW

File: services/common/src/components/documents/DocumentTable.tsx [hunk 1]
```diff
@@ -368,6 +368,7 @@ export const DocumentTable: FC<DocumentTableProps> = ({
     : {};
 
   const coreTableProps = {
+    rowKey: "document_manager_guid",
```
Comment (taraepp) [inline services/common/src/components/documents/DocumentTable.tsx:371]: I got some errors about keys when I uploaded documents so I threw this in. Judging by the updated snap tests, some components were using mine_document_guid.

File: services/minespace-web/src/components/Forms/projects/majorMineApplication/MajorMineApplicationForm.tsx [hunk 2]
```diff
@@ -162,209 +161,201 @@ const MajorMineApplicationForm: React.FC<MajorMineApplicationFormProps> = ({
 
   return (
     <div>
-      <FormWrapper
-        name={FORM.ADD_MINE_MAJOR_APPLICATION}
-        initialValues={{
-          primary_documents: [],
-          spatial_documents: [],
-          supporting_documents: [],
+      <Row>
```
Comment (taraepp) [inline services/minespace-web/src/components/Forms/projects/majorMineApplication/MajorMineApplicationForm.tsx:164]: it thinks this is a large diff but I swear all I did was take out FormWrapper and the handleSubmit prop.

**Review by simensma-fresh — APPROVED** [review]

File: services/minespace-web/src/components/pages/Project/MajorMineApplicationPage.tsx [hunk 3]
```diff
@@ -0,0 +1,359 @@
...
+import MajorMineApplicationCallout from "@/components/Forms/projects/majorMineApplication/MajorMineApplicationCallout";
+import * as routes from "@/constants/routes";
+import { fetchMineDocuments } from "@mds/common/redux/actionCreators/mineActionCreator";
+import Loading from "@mds/common/components/common/Loading";
+
+import FormWrapper from "@mds/common/components/forms/FormWrapper";
+
+export const MAJOR_MINE_APPLICATION_SUBMISSION_STATUSES = ["SUB", "UNR", "APV"];
+
+export const MajorMineApplicationPage: FC = () => {
```
Comment (matbusby-fw) [inline services/minespace-web/src/components/pages/Project/MajorMineApplicationPage.tsx:27]: Love me some TS upgrades!

**Review by matbusby-fw — APPROVED** [review]
Looks great!  Yay dom testing included!

**Review by asinn134 — APPROVED** [review]

**Pull request merged**